src/index.ts

```diff
@@ -27,17 +39,25 @@ interface IParserObject {
 }
 
 /**
- * Parse templateString and substite the given JSON.
+ * Compile template with the given JSON into string or optionally by value.
+ * @param jsonObject a valid parsed JSON object
+ * @param templateString a string that may contain template delimeters `${}` that refer to the JSON.
+ * @param passByValue allows you to return JSON objects by value if there is no `head` or `tail`.
+ * @return a string OR a value if `passByValue` is `true` and there is no `head` or `tail`.
  */
-function parse(jsonObject: any, templateString: string): string {
+function compileTemplate(
+  jsonObject: any,
+  templateString: string,
+  passByValue?: boolean,
```

Comment:
IMO it would be best to make this an options object, as I imagine there will be many others. 

Also, the keynames will help the code be more self documenting. Coming across a third param `true` isn't as helpful as an object, with the key `passByValue` set to true